File: server/bcephal-accounting-service/src/main/java/com/moriset/bcephal/accounting/domain/PostingEntry.java
/**
 * 
 */
package com.moriset.bcephal.accounting.domain;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.Persistent;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 
 * @author MORISET-004
 *
 */

@Entity(name = "PostingEntry")
@Table(name = "BCP_ACCOUNTING_POSTING_ENTRY")
@Data
@EqualsAndHashCode(callSuper = false)

public class PostingEntry extends Persistent{

	/**
	 * 
	 */
	private static final long serialVersionUID = 4629946820510457050L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "posting_ent__seq")
	@SequenceGenerator(name = "posting_ent__seq", sequenceName = "posting_ent__seq", initialValue = 1,  allocationSize = 1)
	private Long id;
	
	@JsonIgnore
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "posting")
	private Posting posting;
	
	private String accountId;
	
	private String accountName;
	
	@Column(name="comment_")
	private String comment;
	
	private String username;
	
	private BigDecimal amount;
	
	private int position;
	
	@Enumerated(EnumType.STRING) 
	@Column(name="sign_")
	private PostingSign sign;
	
	@Enumerated(EnumType.STRING) 
	private PostingStatus status;
	

	@JsonIgnore
	public BigDecimal getSignedAmount() {
		BigDecimal value = getAmount();
		if(getSign() != null && getSign().isDebit()) {
			value = value.negate();
		}
		return value;
	}


	@Override
	public Persistent copy() {
		// TODO Auto-generated method stub
		return null;
	}
	
	
}

File: client/bcephal-client-model/Accounting/PostingEntry.cs
using Bcephal.Models.Accounting;
using Newtonsoft.Json;
using System;

namespace Bcephal.Models.Base.Accounting
{
    public class PostingEntry : Persistent
    {

        public string AccountId { get; set; }

        public string AccountName { get; set; }

        public decimal? Amount { get; set; }

        public int Position { get; set; }

        public string Comment { get; set; }

        public string Username { get; set; }

        public string sign { get; set; }
        [JsonIgnore]
        public PostingSign Sign
        {
            get { return !string.IsNullOrEmpty(sign) ? PostingSign.GetByCode(sign) : null; }
            set { this.sign = value != null ? value.code : null; }
        }

        public string status { get; set; }
        [JsonIgnore]
        public PostingStatus Status
        {
            get { return !string.IsNullOrEmpty(status) ? PostingStatus.GetByCode(status) : PostingStatus.DRAFT; }
            set { this.status = value != null ? value.code : null; }
        }
        [JsonIgnore]
        public int EditPosition { get; set; }

        [JsonIgnore]
        public decimal SignedAmount
        {
            get
            {
                if (!this.Amount.HasValue)
                {
                    this.Amount = decimal.Zero;
                }
                if (this.Sign == PostingSign.DEBIT)
                {
                    return decimal.Zero - this.Amount.Value;
                }
                return this.Amount.Value;
            }
        }


        

        public override String ToString()
        {
            return AccountId;
        }

        public override int CompareTo(object obj)
        {
            if (obj == null || !(obj is PostingEntry)) return 1;
            return this.Position.CompareTo(((PostingEntry)obj).Position);
        }
    }
}
